pkg/build/api/types.go

```diff
@@ -228,8 +228,58 @@ type BuildStatus struct {
 
 	// Output describes the Docker image the build has produced.
 	Output BuildStatusOutput
+
+	// Steps contains details about each build step
+	Steps Steps
+}
+
+// Steps is a slice of build steps that have been recorded.
+type Steps []StepInfo
+
+// RecordInfo records details about each build step
+func (s *Steps) RecordInfo(stepName StepName, startTime unversioned.Time, stopTime unversioned.Time) {
```

Comment:
@csrwng maybe i'm being too javay, but it seems pretty logical to have a method like this on the data structure itself rather than as a helper.  For all the usual data encapsulation/abstraction reasons.

as for where such a definition belongs. personally i prefer to see struct methods defined in the same place as the struct (again, perhaps being too javay), though i can see how that might start to pollute types.go.

is there a reason why we have historically not added heper methods to api types?  it's not like it's part of the public api or anything.